ast/compile.go

```diff
@@ -4744,7 +4744,7 @@ func checkUnusedAssignedVars(loc *Location, stack *localDeclaredVars, used VarSe
 	for v, occ := range dvs.occurrence {
 		// A var that was assigned in this scope must have been seen (used) more than once (the time of assignment) in
 		// the same, or nested, scope to be counted as used.
-		if !v.IsWildcard() && occ == assignedVar && stack.Count(v) <= 1 {
+		if !v.IsWildcard() && stack.Count(v) <= 1 && (occ == assignedVar || occ == argVar) {
```

Comment:
`stack.Count(v) <= 1` holds true for function arguments too as they are rewritten by the compiler and that will be considered as the variable being used(seen). Is this correct @srenatus ?

Reply:
Also if the above it true,then we should probably rename `checkUnusedAssignedVars`

Reply:
Oh, I can't tell from the top of my head... I'd experiment with a test case in the debugger, I think.

Reply:
yeah, `stack.Count(v) <= 1` does the trick for function arguments too if they have seen less than 1 (at least that was what I saw thru the debugger 😸 )

Reply:
any recommendations for the naming? mb we can simplify it as `checkUnusedVars()`

Reply:
`checkUnusedAssignedAndArgVars()` if we want be more specific.

Reply:
I've renamed it to `checkUnusedAssignedAndArgVars()` both in tests and compiler